gradio/sketch/utils.py

```diff
@@ -26,3 +33,95 @@ def set_kwarg(obj: dict, key: str, value: str) -> None:
         obj[key] = ast.literal_eval(value)
     else:
         obj[key] = value
+
+
+def ai(
+    prompt: str,
+    hf_token: str,
+    fn_name: str,
+    inputs: list[tuple[str, type, dict]],
+    output_types: list[tuple[type, dict]],
+):
+    full_prompt = f"""Create a python function with the following header:
+`def {fn_name}({", ".join(i[0] for i in inputs)}):`\n"""
+    if len(inputs) > 0:
+        if len(inputs) == 1:
+            full_prompt += f"""The value of '{inputs[0][0]}' is passed as: {get_value_description(inputs[0][1], inputs[0][2])}.\n"""
+        else:
+            full_prompt += "The inputs are passed as follows:\n"
+            for i in inputs:
+                full_prompt += f"""- '{i[0]}' is passed as: {get_value_description(i[1], i[2])}.\n"""
+    if len(output_types) > 0:
+        if len(output_types) == 1:
+            full_prompt += f"""The function should return a value as: {get_value_description(output_types[0][0], output_types[0][1])}.\n"""
+        else:
+            full_prompt += (
+                "The function should return a tuple of the following values:\n"
+            )
+            for index, o in enumerate(output_types):
+                full_prompt += f"""- index {index} should be: {get_value_description(o[0], o[1])}.\n"""
+    full_prompt += f"""The function should perform the following task: {prompt}\n"""
+    full_prompt += "Return only the python code of the function in your response. Do not wrap the code in backticks or include any description before the response. Return ONLY the function code. Start your response with the header provided. Include any imports inside the function.\n"
+    full_prompt += """If using an LLM would help with the task, use the huggingface_hub library. For example:
```

Comment:
I think it's worth adding the other supported tasks here: https://huggingface.co/docs/huggingface_hub/en/guides/inference#supported-providers-and-tasks

I'm sure people will try things like image generation, which right now causes the generated functions to use obscure dependencies:

<img width="341" alt="image" src="https://github.com/user-attachments/assets/6f5f754b-cb1f-42eb-b1b5-58e40e7bd2fc" />